src/components/input/inputNumber/inputNumber.tsx

```diff
@@ -0,0 +1,204 @@
+
+    const [isFocused, setIsFocused] = useState(false);
+    const {
+        ref: numberMaskRef,
+        value: maskedValue,
+        unmaskedValue,
+        setValue,
+    } = useNumberMask({
+        min,
+        max,
+        value,
+        onChange,
+    });
+
+    const suffixedValue = maskedValue && suffix ? maskedValue + suffix : maskedValue;
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'ArrowUp') {
+            handleIncrement();
+        } else if (e.key === 'ArrowDown') {
+            handleDecrement();
+        }
+
+        onKeyDown?.(e);
+    };
+
+    const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(true);
+        onFocus?.(e);
+    };
+
+    const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(false);
+        onBlur?.(e);
+    };
+
+    const handleIncrement = () => {
+        const parsedValue = Number(unmaskedValue ?? 0);
+
+        // return the input value if it's bigger than the max
+        if (parsedValue > max) {
+            setValue(parsedValue.toString());
+            return;
+        }
+
+        // increment directly to the minimum if value is less than the minimum
+        if (parsedValue < min) {
+            setValue(min.toString());
+            return;
+        }
+
+        // round down to the nearest multiple of the step when the value
+        // is not already a multiple of the step
+        let newValue = parsedValue + step;
+        if (parsedValue % step !== 0) {
+            newValue = Math.floor(parsedValue / step) * step;
+        }
+
+        // increment value with step if it's smaller than the initial value
+        if (newValue < parsedValue) {
+            newValue += step;
+        }
+
+        // ensure the new value is within the min and max range
+        newValue = Math.max(min, Math.min(max, newValue));
+        setValue(newValue.toString());
+    };
+
+    const handleDecrement = () => {
+        const parsedValue = Number(unmaskedValue ?? 0);
+
+        // if the current value is less than the min, don't decrement
+        if (parsedValue < min) {
+            setValue(parsedValue.toString());
+        }
+
+        // decrement value by the step
+        let newValue = parsedValue - step;
+
+        // if the value is not a multiple of the step,
+        // decrement to the biggest multiple of the step that is less than the value
+        if (parsedValue % step !== 0) {
+            newValue = Math.floor(parsedValue / step) * step;
+        }
+
+        // ensure the new value is within the min and max range
+        newValue = Math.max(min, Math.min(max, newValue));
+        setValue(newValue.toString());
+    };
+
+    return (
+        <InputContainer
+            className={classNames('relative', containerClassName)}
+            {...otherContainerProps}
+            isDisabled={isDisabled}
+        >
```

Comment:
The `relative` class is probably not needed here, right? We can just:
```suggestion
        <InputContainer isDisabled={isDisabled} {...otherContainerProps}>
```